File: 2022-04/typescript/739_Daily_Temperatures.ts

function dailyTemperatures(T: Array<number>): Array<number> {
    const answer = new Array(T.length).fill(0)
    const stack = []

    T.forEach((v, k) => {
        while (stack.length > 0 && v > T[stack[stack.length - 1]]) {
            const index = stack.pop()
            answer[index] = k - index
        }
        stack.push(k)
    })

    return answer
}

File: 2022-04/python/739_Daily_Temperatures.py

class Solution:

    def dailyTemperatures(self, T: List[int]) -> List[int]:
        answer = [0] * len(T)
        s = []
        for i, t in enumerate(T):
            while s and t > T[s[-1]]:
                idx = s.pop()
                answer[idx] = i - idx
            s.append(i)
        return answer